### PR TITLE
Mitigate OAuth2Client reapply

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Adds metadata: null  and finilizers fields to chart, which clean up caused  errors in hydra-maester

### CODE REVIEW

File: resources/eventing/charts/controller/templates/oauth2-client.yaml [hunk 1]
```diff
@@ -4,9 +4,12 @@ kind: OAuth2Client
 metadata:
   name: {{ include "controller.fullname" . }}
   labels: {{- include "controller.labels" . | nindent 4 }}
+  finalizers:
```
Comment (jakobmoellersap) [inline resources/eventing/charts/controller/templates/oauth2-client.yaml:7]: I suggest to never ever add finalisers coming from a controller since it will block deletion of a resource if the controller is gone. Rather the edited resource should be patched via strategic merge, ignoring the finalisers. I suggest to review the logic that forces this field to be present

Comment (barchw) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml:7]: Would it be possible then to configure reconciler not to remove the `finalizers` if they were applied by hydra-maester controller? For the sake of fixing the error that would also be OK

Comment (k15r) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml:7]: 😕 reconciler removes finalizers? In what scenario?

Comment (jakobmoellersap) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml:7]: TBH also surprising for me. Finalisers outside of the reconciler should be ignored imo. Also setting nil metadata would result in them being removed on every reconcile. @tobiscr are you aware of the behaviour here? I almost cannot imagine that this never was an issue before...

Comment (tobiscr) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml:7]: The finalisers are ignored by the old reconciler until we are deleting the cluster. During the deletion, one of the final steps is the cleanup of finalisers to make sure every resources gets dropped (just to make sure that there are no "dead bodies" left).

Comment (tobiscr) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml:7]: If you have to deal with finalisers during runtime, you can either handle such cases in a custom reconciler (but this will update finalisers only during the regular reconciliation runs) or you handle it over an operator during runtime.

Comment (barchw) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml]: Removed the finalizers as those will be handled by https://github.com/kyma-incubator/reconciler/pull/1274

File: resources/eventing/charts/controller/templates/oauth2-client.yaml [hunk 2]
```diff
@@ -8,5 +8,6 @@ spec:
   grantTypes:
     - client_credentials
   hydraAdmin: {}
+  metadata: null
```
Comment (k15r) [inline resources/eventing/charts/controller/templates/oauth2-client.yaml:11]: Is this change really necessary?

Comment (k15r) [reply resources/eventing/charts/controller/templates/oauth2-client.yaml:11]: well then ill approve this noop change :D

**Review by k15r — APPROVED** [review]
